playbooks/troubleshooting/troubleshooting_guide.adoc

```diff
@@ -0,0 +1,719 @@
+
+----
+[root@master00 ~]# oc describe pod docker-registry-2-n8d21
+
+time="2016-03-10T09:00:56.671348073-05:00" level=error msg="response completed with error" err.code="BLOB_UPLOAD_INVALID" err.detail="Invalid token" err.message="blob upload invalid" go.version=go1.4.2 http.request.host="172.50.225.185:5000" http.request.id=e4066c94-950d-4306-89de-57a1ac573f72 http.request.method=PUT http.request.remoteaddr="10.5.0.1:34874" http.request.uri="/v2/ex3/tm/blobs/uploads/11158157-1eb4-4ba6-9327-9e01a8cbc103?_state=HXJVBhFZdeHo5zeLrzyKKMGb7NPxCQq-Fawt-zNaYBB7Ik5hbWUiOiJleDMvdG0iLCJVVUlEIjoiMTExNTgxNTctMWViNC00YmE2LTkzMjctOWUwMWE4Y2JjMTAzIiwiT2Zmc2V0Ijo1MjA2MjIwODAsIlN0YXJ0ZWRBdCI6IjIwMTYtMDMtMTBUMTM6NTc6NDNaIn0%3D&digest=sha256%3Ab30d0a02a4a259346c94eca8c6150b48a2132cf6821332e3196f2cfe0316d42b" http.request.useragent="docker/1.8.2-el7 go/go1.4.2 kernel/3.10.0-327.10.1.el7.x86_64 os/linux arch/amd64" http.response.contenttype="application/json; charset=utf-8" http.response.duration=180.043287ms http.response.status=404 http.response.written=88 instance.id=79ab5634-8822-4e05-95b7-f13c42fee017 vars.name="ex3/tm" vars.uuid=11158157-1eb4-4ba6-9327-9e01a8cbc103
+
+10.5.0.1 - - [10/Mar/2016:09:00:56 -0500] "PUT /v2/ex3/tm/blobs/uploads/11158157-1eb4-4ba6-9327-9e01a8cbc103?_state=HXJVBhFZdeHo5zeLrzyKKMGb7NPxCQq-Fawt-zNaYBB7Ik5hbWUiOiJleDMvdG0iLCJVVUlEIjoiMTExNTgxNTctMWViNC00YmE2LTkzMjctOWUwMWE4Y2JjMTAzIiwiT2Zmc2V0Ijo1MjA2MjIwODAsIlN0YXJ0ZWRBdCI6IjIwMTYtMDMtMTBUMTM6NTc6NDNaIn0%3D&digest=sha256%3Ab30d0a02a4a259346c94eca8c6150b48a2132cf6821332e3196f2cfe0316d42b HTTP/1.1" 404 88 "" "docker/1.8.2-el7 go/go1.4.2 kernel/3.10.0-327.10.1.el7.x86_64 os/linux arch/amd64"
+----
+
+==== Solution
+
+There is a https://bugzilla.redhat.com/show_bug.cgi?id=1277356[Red Hat
+Bugzilla report] describing that the solution is to add **no_wdelay** to
+the nfs export options:
+
+----
+(rw,sync,root_squash,no_wdelay)
+----
+
+*OR* add session affinity to the registry service:
+
+----
+oc get -o yaml service docker-registry \| \
+      sed 's/\(sessionAffinity:\s*\).*/\1ClientIP/' \| \
+      oc replace -f -
+----
+
+Restart the nfs server and restart the S2I build
+ +
+ +
+
+== Quotas and Limitranges
+
+=== Must make a non-zero request for cpu
+
+After creating a quota for for CPU usage inside of a project you receive the
+following error:
+----
+failed to create build pod: Pod "nodejs-example-2-build" is forbidden: must make a non-zero request for cpu since it is tracked by quota.
+----
+
+==== Solution:
+
+There must be a corresponding CPU amount defined in your project limits
+to go along with the defined quota. Click for more information on
+https://docs.openshift.com/enterprise/3.1/dev_guide/quota.html[quotas]
+and
+https://docs.openshift.com/enterprise/3.1/dev_guide/limits.html[limits].
+ +
+ +
+
+=== I have enough ram for another pod but it won't build
+
+Quotas can prevent build if *build machine* + *new pod* is greater than
```

Comment:
Can you expand this one to show a behavior, and a solution of how to fix it? Maybe a link out to quotas documentation.